Druid/DRInstruction.class.st

```diff
@@ -112,8 +112,11 @@ DRInstruction >> breakBasicBlock [
 { #category : #'loop-invariance' }
 DRInstruction >> canMoveOutsideLoop: loop [
 
-	^ self operands allSatisfy: [ :operand |
-		operand isInvariantFrom: loop ]
+	^ (self operands allSatisfy: [ :operand | operand isInvariantFrom: loop ]) and: [ 
+		"Can not be inside a if (inside the loop). 
+		In that case also depends of conditional jump operands"
+		  (self controlFlowGraph allBlocksBetween: loop headerBlock and: self basicBlock)
```

Comment:
Maybe later this can be renamed as "body" ?

Reply:
That already exists in `DRLoop`, this could be written based on that. Also, be improved, I don't like this definition so much (I don't remember why I changed it).